corehq/apps/app_manager/views/utils.py

```diff
@@ -437,3 +438,41 @@ def get_multimedia_sizes_for_build(domain, build_id, build_profile_id=None):
         media_object = media_objects[multimedia_id]
         total_size[media_object.doc_type] += media_object.content_length
     return total_size
+
+
+def handle_shadow_child_modules(app, shadow_parent):
+    """Creates or deletes shadow child modules if the parent module requires
+
+    Used primarily when changing the "source module id" of a shadow module
+    """
+    source_module_children = [
+        m for m in app.modules
+        if m.root_module_id == shadow_parent['source_module_id']
+    ]
+
+    shadow_parent_children = [
+        m for m in app.modules
+        if m.root_module_id == shadow_parent.unique_id
+    ]
+    changes = False
+
+    # Delete unneeded modules
+    for child in shadow_parent_children:
+        if child.source_module_id not in source_module_children:
+            changes = True
+            app.delete_module(child.unique_id)
+
+    # Add new modules
+    for shadow_child in source_module_children:
+        changes = True
+        new_shadow = ShadowModule.new_module(shadow_child.name['en'], 'en')
+        new_shadow.source_module_id = shadow_child.unique_id
+        new_shadow.root_module_id = shadow_parent['unique_id']
+        new_shadow.put_in_root = shadow_child.put_in_root
+        app.add_module(new_shadow)
+
+    if changes:
+        app.move_child_modules_after_parents()
```

Comment:
I'm not sure if calling this is "allowed" as it seems to potentially change an app's behaviour / could break custom detail screens, etc which have hard-coded module ids.
The alternative is to make the changes as above, but then show people the "Your modules have been reordered" message as we did before the migration.
I'd be interested in your thoughts @esoergel